src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -94,6 +95,13 @@ public static class TagHelperBlockRewriter
             return attributes;
         }
 
+        private static bool IsSelfClosing(Block beginTagBlock)
+        {
+            var childSpan = beginTagBlock.Children.Last() as Span;
```

Comment:
use `beginTagBlock.FindLastDescendentSpan()` to avoid problems when the tree is more than one level deep